drivers/timer/stm32_lptim_timer.c

```diff
@@ -0,0 +1,285 @@
+
+		/* announce the elapsed time in ms, cnt_value is 16bit */
+		u32_t dticks = (cnt_value
+				* (CONFIG_SYS_CLOCK_TICKS_PER_SEC))
+				/ (LPTIM_CLOCK);
+
+		z_clock_announce(IS_ENABLED(CONFIG_TICKLESS_KERNEL)
+			? dticks : 1);
+	}
+}
+
+int z_clock_driver_init(struct device *device)
+{
+	ARG_UNUSED(device);
+
+	/* enable LPTIM clock source */
+	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_LPTIM1);
+	LL_APB1_GRP1_ReleaseReset(LL_APB1_GRP1_PERIPH_LPTIM1);
+
+#if defined(CONFIG_STM32_LPTIM_CLOCK_LSI)
+	/* enable LSI clock */
+	LL_RCC_LSI_Enable();
+	while (!LL_RCC_LSI_IsReady()) {
+		/* Wait for LSI ready */
+	}
+	LL_RCC_SetLPTIMClockSource(LL_RCC_LPTIM1_CLKSOURCE_LSI);
+
+#else /* CONFIG_STM32_LPTIM_CLOCK_LSI */
+	/* Enable the power interface clock */
+	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_PWR);
+	/* enable backup domain */
+	LL_PWR_EnableBkUpAccess();
+	LL_RCC_ForceBackupDomainReset();
+	LL_RCC_ReleaseBackupDomainReset();
+
+	/* enable LSE clock */
+	LL_RCC_LSE_DisableBypass();
+	LL_RCC_LSE_Enable();
+	while (!LL_RCC_LSE_IsReady()) {
+		/* Wait for LSE ready */
+	}
+	LL_RCC_SetLPTIMClockSource(LL_RCC_LPTIM1_CLKSOURCE_LSE);
+
+#endif /* CONFIG_STM32_LPTIM_CLOCK_LSI */
+
+	/* Clear the event flag and possible pending interrupt */
+	IRQ_CONNECT(DT_LPTIM_1_IRQ, DT_LPTIM_1_IRQ_PRI,
+			lptim_irq_handler, 0, 0);
+	irq_enable(DT_LPTIM_1_IRQ);
+
+	/* configure the LPTIM1 counter */
+	LL_LPTIM_SetClockSource(LPTIM1, LL_LPTIM_CLK_SOURCE_INTERNAL);
+	/* configure the LPTIM1 prescaler with 1 */
+	LL_LPTIM_SetPrescaler(LPTIM1, LL_LPTIM_PRESCALER_DIV1);
+	LL_LPTIM_SetPolarity(LPTIM1, LL_LPTIM_OUTPUT_POLARITY_REGULAR);
+	LL_LPTIM_SetUpdateMode(LPTIM1, LL_LPTIM_UPDATE_MODE_IMMEDIATE);
+	LL_LPTIM_SetCounterMode(LPTIM1, LL_LPTIM_COUNTER_MODE_INTERNAL);
+	LL_LPTIM_DisableTimeout(LPTIM1);
+	/* counting start is initiated by software */
+	LL_LPTIM_TrigSw(LPTIM1);
+	/* default input source configuration */
+	LL_LPTIM_SetInput1Src(LPTIM1, LL_LPTIM_INPUT1_SRC_GPIO);
+	LL_LPTIM_SetInput2Src(LPTIM1, LL_LPTIM_INPUT2_SRC_GPIO);
+
+	/* LPTIM1 interrupt set-up before enabling */
+	/* no Compare match Interrupt */
+	LL_LPTIM_DisableIT_CMPM(LPTIM1);
+	LL_LPTIM_ClearFLAG_CMPM(LPTIM1);
+
+	/* Autoreload match Interrupt */
+	LL_LPTIM_EnableIT_ARRM(LPTIM1);
+	LL_LPTIM_ClearFLAG_ARRM(LPTIM1);
+
+	last_cnt = 0;
+	accumulated_lptim_ms = 0;
+
+	if (IS_ENABLED(CONFIG_TICKLESS_KERNEL)) {
+		/* Enable the LPTIM1 counter */
+		LL_LPTIM_Enable(LPTIM1);
+
+		/* Set the Autoreload value once the timer is enabled */
+		LL_LPTIM_SetAutoReload(LPTIM1, LPTIM_TIMEBASE);
+
+		/* do not use compare yet, set the compare to period */
+		LL_LPTIM_SetCompare(LPTIM1, 0);
+
+		/* Start the LPTIM counter in continuous mode */
+		LL_LPTIM_StartCounter(LPTIM1,
+				LL_LPTIM_OPERATING_MODE_CONTINUOUS);
+	} else {
+		/* Enable the LPTIM1 counter */
+		LL_LPTIM_Enable(LPTIM1);
+
+		/* Set the Autoreload value once the timer is enabled */
+		LL_LPTIM_SetAutoReload(LPTIM1, COUNT_PER_TICK);
```

Comment:
The 2 different initialization could be worth a few comments.
so in case of TICKS mode, the auto reload is set to COUNT_PER_TICK so there will be an interrupt for each tick and the counter will always reset right ?  and the comparator will always be kept to 0 and never modified ?